hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/transfer/ClassicTransfersTranslator.java

```diff
@@ -87,50 +82,43 @@ public ClassicTransfersCall callFrom(@NonNull final HtsCallAttempt attempt) {
                 // test would use the qualified delegate id if applicable
                 // test would use the qualified delegate id if applicable.
                 // Only certain functions support qualified delegates, so restrict to those.
-                isClassicCallSupportingQualifiedDelegate(selector) ? attempt.authorizingId() : attempt.senderId(),
+                isClassicCallSupportingQualifiedDelegate(attempt) ? attempt.authorizingId() : attempt.senderId(),
                 decoder.checkForFailureStatus(attempt),
                 nominalBodyFor(attempt),
                 attempt.configuration(),
-                isClassicCall(selector) ? APPROVAL_SWITCH_HELPER : null,
+                isClassicCall(attempt) ? APPROVAL_SWITCH_HELPER : null,
                 CALL_STATUS_STANDARDIZER,
                 attempt.defaultVerificationStrategy(),
                 SYSTEM_ACCOUNT_CREDIT_SCREEN,
                 SPECIAL_REWARD_RECEIVERS);
     }
 
     private @Nullable TransactionBody nominalBodyFor(@NonNull final HtsCallAttempt attempt) {
```

Comment:
This method is curious.  Almost suggests an `AbstractCallAttempt.indexOfSelector()` which returns the _index_ of the searched for selector in the list (like `List.indexOf`) which you could then use to index a parallel list of method references.

Maybe even simpler would be a `Map` of selector to method reference?